main.go

```diff
@@ -14,191 +14,336 @@ import (
-		time.Sleep(t)
+		for _, stat := range stats {
+			cpuLimitVal := prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Namespace: "mesos_task",
+					Name:      "cpu_limit",
+					Help:      "Fractional CPU limit.",
+					ConstLabels: prometheus.Labels{
+						"task":         stat.Source,
+						"mesos_slave":  host,
+						"framework_id": stat.FrameworkId,
+					},
+				},
+			)
+			cpuLimitVal.Set(stat.Statistics.CpusSystemTimeSecs)
+
+			cpuSysVal := prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Namespace: "mesos_task",
+					Name:      "cpu_system_seconds_total",
+					Help:      "Cumulative system CPU time in seconds.",
+					ConstLabels: prometheus.Labels{
+						"task":         stat.Source,
+						"mesos_slave":  host,
+						"framework_id": stat.FrameworkId,
+					},
+				},
+			)
+			cpuSysVal.Set(stat.Statistics.CpusSystemTimeSecs)
+
+			cpuUsrVal := prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Namespace: "mesos_task",
+					Name:      "cpu_user_seconds_total",
+					Help:      "Cumulative user CPU time in seconds.",
+					ConstLabels: prometheus.Labels{
+						"task":         stat.Source,
+						"mesos_slave":  host,
+						"framework_id": stat.FrameworkId,
+					},
+				},
+			)
+			cpuUsrVal.Set(stat.Statistics.CpusUserTimeSecs)
+
+			memLimitVal := prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Namespace: "mesos_task",
+					Name:      "memory_limit_bytes",
+					Help:      "Task memory limit in bytes.",
+					ConstLabels: prometheus.Labels{
+						"task":         stat.Source,
+						"mesos_slave":  host,
+						"framework_id": stat.FrameworkId,
+					},
+				},
+			)
+			memLimitVal.Set(float64(stat.Statistics.MemLimitBytes))
+
+			memRssVal := prometheus.NewGauge(
+				prometheus.GaugeOpts{
+					Namespace: "mesos_task",
+					Name:      "memory_rss_bytes",
+					Help:      "Task memory RSS usage in bytes.",
+					ConstLabels: prometheus.Labels{
+						"task":         stat.Source,
+						"mesos_slave":  host,
+						"framework_id": stat.FrameworkId,
+					},
+				},
+			)
+			memRssVal.Set(float64(stat.Statistics.MemRssBytes))
+
+			outCh <- cpuLimitVal
+			outCh <- cpuSysVal
+			outCh <- cpuUsrVal
+			outCh <- memLimitVal
+			outCh <- memRssVal
+		}
 	}
 }
 
-func processRequests(c *Config) {
-	for _, endpoint := range *c.Endpoints {
-		glog.Infof("Found %s\n", endpoint.Name)
+func (e *PeriodicExporter) lockSlaves(f func() []string) []string {
+	e.slaves.Lock()
+	defer e.slaves.Unlock()
+	return f()
+}
 
-		if len(endpoint.Interval) == 0 {
-			endpoint.Interval = c.Interval
-		}
+func (e *PeriodicExporter) rLockMetrics(f func()) {
+	e.RLock()
+	defer e.RUnlock()
+	f()
+}
 
-		if len(endpoint.TimeoutStr) != 0 {
-			t, err := time.ParseDuration(endpoint.TimeoutStr)
-			if err != nil {
-				glog.Fatalf("Invalid specification of time.Duration: %s\n", err)
-			}
-			endpoint.Timeout = t
-		} else {
-			endpoint.Timeout = 3 * time.Second
+func (e *PeriodicExporter) runEvery(f func(), interval time.Duration) {
```

Comment:
Since it doesn't use any fields of `e`, this would better be just a general function, not a method.